src/components/WidgetPropertiesForm/index.tsx

```diff
@@ -72,23 +81,73 @@ export default function WidgetPropertiesForm({ id, onClose }: WidgetPropertiesFo
     },
   });
 
-  const schema = useMemo(() => {
-    const translatedSchema =
-      verna.configSchema?.properties?.[templateName] || ({} as VernaJSONSchemaType);
+  useEffect(() => {
+    setTranslatedSchema(translateSchema());
+  }, [verna.locale]);
+
+  function translateSchema() {
+    // Get the schema corresponding to the configuration of this widget and clone its
+    const widgetSchema = _.cloneDeep(verna.configSchema?.properties?.[templateName]);
 
     const isMessageKey = (key: PropertyKey): key is keyof typeof messages => {
       return key in messages;
     };
 
-    if (translatedSchema.properties) {
-      for (const [key, value] of Object.entries(translatedSchema.properties)) {
-        if (isMessageKey(key)) {
-          value.title = formatMessage(messages[key]);
-        }
+    // Translate parameters names
+    for (const [key, value] of Object.entries(widgetSchema?.properties || {})) {
+      if (isMessageKey(key)) {
+        value.title = formatMessage({
+          defaultMessage:
+            (verna.locale && verna.schemaTranslations[verna.locale][key]) ||
+            messages[key].defaultMessage,
+          id: messages[key].id,
+        });
```

Comment:
I'm not a fan this way of doing things. Currently this is working because you are updating an object but this will not work with a literal.